llvm/include/llvm/ADT/GenericCycleInfo.h

```diff
@@ -107,6 +107,12 @@ template <typename ContextT> class GenericCycle {
     return is_contained(Entries, Block);
   }
 
+  /// \brief Replace all entries with \p Block as single entry.
+  void setSingleEntry(BlockT *Block) {
+    Entries.clear();
+    Entries.push_back(Block);
```

Comment:
Could this be `nullptr`? Should we add an assert to guard against this?

Reply:
Good catch! Replaced it with a stronger test to check that `Block` is contained in the current cycle.

Reply:
Fixed.